test/runtests.jl

```diff
@@ -176,6 +176,8 @@ end
         @test y >= x
         @test x == Y(i)
         @test y == X(j)
+        @test isodd(x) == isodd(i) == !iseven(x)
+        @test isodd(y) == isodd(j) == !iseven(y)
```

Comment:
This tests the functionality, which is good, but as this is a performance PR, you might also check that it doesn't allocate, like `@test @allocated isodd(x) == 0`.

Reply:
also, it would be good to put the test in the same section as in the source definition.

Reply:
Cool, done both of those! :) Thanks!